### PR TITLE
Bug 1862848: Empty OpenStack "zones" property

### PR DESCRIPTION
**Pull request by pierreprinetti**
Before this patch, manifest generation would try a divide by zero if no
Availability Zones were passed in the OpenStack platform spec.

### CODE REVIEW

File: pkg/asset/machines/openstack/machines_test.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	return &types.InstallConfig{
+		AdditionalTrustBundle: "",
+		SSHKey:                "",
+		BaseDomain:            "basedomain.example.com",
+		Networking:            nil,
+		ControlPlane:          nil,
+		Compute:               nil,
+		Platform: types.Platform{
+			OpenStack: &openstackTypes.Platform{
+				Region:                 "RegionOne",
```
Comment (Fedosin) [inline pkg/asset/machines/openstack/machines_test.go:23]: Region is deprecated along with TrunkSupport and OctaviaSupport

File: pkg/asset/machines/master.go [hunk 2]
```diff
@@ -238,7 +238,12 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 
 		imageName, _ := rhcosutils.GenerateOpenStackImageName(string(*rhcosImage), clusterID.InfraID)
 
-		machines, err = openstack.Machines(clusterID.InfraID, ic, pool, imageName, "master", "master-user-data-managed")
+		trunkSupport, err := openstack.CheckNetworkExtensionAvailability(ic.Platform.OpenStack.Cloud, "trunk")
```
Comment (Fedosin) [inline pkg/asset/machines/master.go:241]: What was the reason to move `CheckNetworkExtensionAvailability` function call here and add `trunkSupport` parameter to `Machines`?

Comment (pierreprinetti) [reply pkg/asset/machines/master.go:241]: By moving the call to `checkNetworkExtensionAvailability` outside of `func Machine`, we can now unit-test `func Machines` without any call to the cloud.

File: pkg/asset/machines/openstack/machines_test.go (hunk 1 above)
Comment (pierreprinetti) [reply pkg/asset/machines/openstack/machines_test.go:23]: oh I missed that, thanks! If it's OK for you, I'll comment out `Region` in a follow-up; there's a couple more tests I want to add but didn't want to clutter this urgent PR.

File: pkg/asset/machines/master.go (hunk 2 above)
Comment (Fedosin) [reply pkg/asset/machines/master.go:241]: ah, yes... it's okay then :)

**Comment by iamemilio** [comment]
@pierreprinetti nice work, the unit tests for machines will go a long way

**Comment by Fedosin** [comment]
/lgtm cancel

It turned out that by design Zones can't be empty. There is another bug that prevents setting right defaults for this parameter: https://github.com/openshift/installer/pull/4004

**Comment by pierreprinetti** [comment]
We might want to merge this anyway, after #4004 is in. WDYT @Fedosin ?

**Comment by pierreprinetti** [comment]
/close
Superseded by #4004

**Pull request closed without merging**